src/Aspire.Hosting.Sdk/SDK/Sdk.in.targets

```diff
@@ -71,13 +71,17 @@
       <_AppHostVersion>%(_AppHostPackageReference.Version)</_AppHostVersion>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(_AppHostVersion)' != ''">
-      <__CurrentAppHostVersionMessage> You are using version $(_AppHostVersion).</__CurrentAppHostVersionMessage>
+    <PropertyGroup Condition="'$(_AppHostVersion)' != '' and $([MSBuild]::VersionLessThan('$(_AppHostVersion)', '8.2.0'))">
+      <!-- If we find the version to Aspire.Hosting.AppHost package but it is lower than 8.2.0, then we fall back
+      to use the Dashboard and DCP packages that match the version of the installed workload for backwards compatibility.
+      This results in the same behavior that we had before moving Dashboard and DCP out of the workload, since the version
+      is again just matching to the one the workload has. -->
+      <_AppHostVersion>@VERSION@</_AppHostVersion>
     </PropertyGroup>
 
     <!-- At this point, we should have the version either by CPM or PackageReference, so we fail if not.  -->
-    <Error Condition="'$(_AppHostVersion)' == '' or $([MSBuild]::VersionLessThan('$(_AppHostVersion)', '8.2.0'))" 
-       Text="$(MSBuildProjectName) is a .NET Aspire AppHost project that needs a package reference to Aspire.Hosting.AppHost version 8.2.0 or above to work correctly.$(__CurrentAppHostVersionMessage)" />
+    <Error Condition="'$(_AppHostVersion)' == ''" 
```

Comment:
A test for this would be good too. It doesn't have to be in this PR.
Also, the same for both cases but with CPM?

Reply:
I'll add more tests to this whole scenario with the changes on the Aspire Workload for 9.0